src/Datadog.Trace/Configuration/StringConfigurationSource.cs

```diff
@@ -17,6 +17,18 @@ public abstract class StringConfigurationSource : IConfigurationSource
         /// <param name="data">A string containing key-value pairs which are comma-separated, and for which the key and value are colon-separated.</param>
         /// <returns><see cref="IDictionary{TKey, TValue}"/> of key value pairs.</returns>
         public static IDictionary<string, string> ParseCustomKeyValues(string data)
+        {
+            return ParseCustomKeyValues(data, allowOptionalMappings: false);
+        }
+
+        /// <summary>
+        /// Returns a <see cref="IDictionary{TKey, TValue}"/> from parsing
+        /// <paramref name="data"/>.
+        /// </summary>
+        /// <param name="data">A string containing key-value pairs which are comma-separated, and for which the key and value are colon-separated.</param>
+        /// <param name="allowOptionalMappings">Determines whether to create dictionary entries when the input has no value mapping</param>
+        /// <returns><see cref="IDictionary{TKey, TValue}"/> of key value pairs.</returns>
+        public static IDictionary<string, string> ParseCustomKeyValues(string data, bool allowOptionalMappings = false)
```

Comment:
I looks like we can make `allowOptionalMappings` non optional since we have the overload without it. (I thought this generated a compiler warning, but maybe it's just R#)

Reply:
Good idea, done in c1ac5ce1868b7c359e741bd04d05e742647a9a33